src/unk_0203E880.c

```diff
@@ -739,84 +603,74 @@ static u16 Unk_02100AD4[] = {
     }
 
-    v0 = (LCRNG_Next() % (NELEMS(Unk_02100AE0)));
+    rand = LCRNG_Next() % NELEMS(sIronIslandHiddenItemFlags);
 
-    if (fieldSystem->location->mapId != Unk_02100AE0[v0][0]) {
-        sub_0203F1B0(fieldSystem, (730 + Unk_02100AE0[v0][1]));
+    if (fieldSystem->location->mapId != sIronIslandHiddenItemFlags[rand][0]) {
+        FieldSystem_ClearFlag(fieldSystem, FLAG_OFFSET_HIDDEN_ITEMS + sIronIslandHiddenItemFlags[rand][1]);
     }
 
-    if (fieldSystem->location->mapId != 256) {
-        v0 = (LCRNG_Next() % (NELEMS(Unk_02100AD4)));
-        sub_0203F1B0(fieldSystem, (730 + Unk_02100AD4[v0]));
+    if (fieldSystem->location->mapId != MAP_HEADER_FLOAROMA_MEADOW) {
+        rand = LCRNG_Next() % NELEMS(sFloaromaMeadowHiddenItemFlags);
+        FieldSystem_ClearFlag(fieldSystem, FLAG_OFFSET_HIDDEN_ITEMS + sFloaromaMeadowHiddenItemFlags[rand]);
 
-        v0 = (LCRNG_Next() % (NELEMS(Unk_02100AD4)));
-        sub_0203F1B0(fieldSystem, (730 + Unk_02100AD4[v0]));
+        rand = LCRNG_Next() % NELEMS(sFloaromaMeadowHiddenItemFlags);
+        FieldSystem_ClearFlag(fieldSystem, FLAG_OFFSET_HIDDEN_ITEMS + sFloaromaMeadowHiddenItemFlags[rand]);
     }
-
-    return;
 }
 
-u8 sub_0203F3C8 (u16 param0)
+u8 Script_GetHiddenItemRange (u16 scriptID)
 {
-    int v0;
-    u16 v1;
-    const HiddenItem * v2;
+    int i;
+    const HiddenItem *item = &gHiddenItems[0];
+    u16 itemScript = Script_GetHiddenItemScript(scriptID);
 
-    v2 = &gHiddenItems[0];
-    v1 = sub_0203F304(param0);
-
-    for (v0 = 0; v0 < (NELEMS(gHiddenItems)); v0++) {
-        if (v2[v0].script == v1) {
+    for (i = 0; i < NELEMS(gHiddenItems); i++) {
+        if (item[i].script == itemScript) {
             break;
         }
     }
 
-    if (v0 >= (NELEMS(gHiddenItems))) {
+    if (i >= NELEMS(gHiddenItems)) {
         GF_ASSERT(FALSE);
         return 0;
     }
 
-    return v2[v0].range;
+    return item[i].range;
 }
 
-static BOOL sub_0203F404 (UnkStruct_0203EF60 * param0, u16 param1)
+static BOOL ScriptManager_SetHiddenItem (ScriptManager * scriptManager, u16 scriptID)
 {
-    int v0;
-    u16 v1;
-    const HiddenItem * v2;
-    u16 * v3 = sub_0203EF60(param0, 41);
-    u16 * v4 = sub_0203EF60(param0, 42);
-    u16 * v5 = sub_0203EF60(param0, 43);
-
-    v2 = &gHiddenItems[0];
-    v1 = sub_0203F304(param1);
+    int i;
+    u16 *scriptParam0 = ScriptManager_GetMemberPtr(scriptManager, SCRIPT_DATA_PARAMETER_0);
+    u16 *scriptParam1 = ScriptManager_GetMemberPtr(scriptManager, SCRIPT_DATA_PARAMETER_1);
+    u16 *scriptParam2 = ScriptManager_GetMemberPtr(scriptManager, SCRIPT_DATA_PARAMETER_2);
+    const HiddenItem *item = &gHiddenItems[0];
+    u16 itemScript = Script_GetHiddenItemScript(scriptID);
 
-    for (v0 = 0; v0 < (NELEMS(gHiddenItems)); v0++) {
-        if (v2[v0].script == v1) {
+    for (i = 0; i < NELEMS(gHiddenItems); i++) {
+        if (item[i].script == itemScript) {
```

Comment:
**polish:** I assume that this does not match if you attempt to index against `gHiddenItems`?

Reply:
Unfortunately, yup, doesn't match.